internal/metrics/proxy_metrics.go

```diff
@@ -0,0 +1,296 @@
+
+	// ProxyInsertVectors record the number of vectors insert successfully.
+	ProxyInsertVectors = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.ProxyRole,
+			Name:      "insert_vectors",
+			Help:      "The number of vectors insert successfully",
+		}, []string{"client_id"})
+
+	// ProxyLinkedSDKs record The number of SDK linked proxy.
+	// TODO: how to know when sdk disconnect?
+	ProxyLinkedSDKs = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.ProxyRole,
+			Name:      "linked_SDK_numbers",
+			Help:      "The number of SDK linked proxy",
+		}, []string{"client_id"})
+
+	// ProxySearchLatency record the latency of search successfully.
+	ProxySearchLatency = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.ProxyRole,
+			Name:      "search_latency",
+			Help:      "The latency of search successfully",
+		}, []string{"client_id", "type"})
+
+	// ProxySendMessageTime record the time that the proxy sent the search request to the message stream.
+	ProxySendMessageTime = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.ProxyRole,
+			Name:      "send_search_msg_time",
+			Help:      "The time that the proxy sent the search request to the message stream",
+		}, []string{"client_id", "type"})
+
+	// ProxyWaitForSearchResultTime record the time that the proxy waits for the search result.
+	ProxyWaitForSearchResultTime = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.ProxyRole,
+			Name:      "wait_for_search_result_time",
+			Help:      "The time that the proxy waits for the search result",
+		}, []string{"client_id", "type"})
+
+	// ProxyReduceSearchResultTime record the time that the proxy reduces search result.
+	ProxyReduceSearchResultTime = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.ProxyRole,
+			Name:      "reduce_search_result_time",
+			Help:      "The time that the proxy reduces search result",
+		}, []string{"client_id", "type"})
+
+	// ProxyDecodeSearchResultTime record the time that the proxy decodes the search result.
+	ProxyDecodeSearchResultTime = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.ProxyRole,
+			Name:      "decode_search_result_time",
+			Help:      "The time that the proxy decodes the search result",
+		}, []string{"client_id", "type"})
+
+	// ProxyMsgStreamObjectsPerPChaPerCollection record the number of MsgStream objects per PChannel on each collection on Proxy.
+	ProxyMsgStreamObjectsPerPChaPerCollection = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.ProxyRole,
+			Name:      "MsgStream_objects_per_PChan_per_Collection",
```

Comment:
name too long.
prometheus will identify this metric as 
<namespace>_<subsystem>_<name>_<type>